os/arch/arm/src/armv7-m/up_sigdeliver.c

```diff
@@ -148,6 +149,14 @@ void up_sigdeliver(void)
 	irqrestore((uint16_t)regs[REG_PRIMASK]);
 #endif
 
+#ifndef CONFIG_SUPPRESS_INTERRUPTS
```

Comment:
I think this change is not specific to porting cortex M7 or imxrt. 
Hence, it must not be part of this commit.